detox/src/devices/drivers/android/emulator/EmulatorDriver.js

```diff
@@ -63,6 +58,10 @@ class EmulatorDriver extends AndroidDriver {
     return adbName;
   }
 
+  async doAllocateDevice(deviceQuery) {
+    return await this.freeDeviceFinder.findFreeDevice(deviceQuery);
```

Comment:
```suggestion
    return await this.freeDeviceFinder.findFreeDevice(deviceQuery) || this._createDevice();
```

Reply:
Good point - `createDevice()` is relevant for emulators only as it synth's an emulator's name with a randomly allocated port